### PR TITLE
[WIP] Release bugfix for logging security

### PR DESCRIPTION
**Pull request by dschaper**
**By submitting this pull request, I confirm the following**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [ ] I have written tests and verified that they fail without my change.
- [x] I have squashed any insignificant commits.
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership. It is compatible with the EUPL 1.2 license.
- [x] I have Signed Off all commits. (`git commit --signoff`)

***Please explain what you have done and wish to accomplish with this Pull Request***

1. What does this change do, exactly?
Implements temporary files for log writing and copies those files to `/var/log/pihole` when complete

2. Please link to the relevant issues.
#1699 
#1764 

3. Which documentation changes (if any) need to be made because of this PR?
Self-documenting

Signed-off-by: Dan Schaper dan.schaper@pi-hole.net

### CODE REVIEW

**Comment by dschaper** [comment]
- [x] basic-install
- [ ] debug
- [ ] ftl log location

(From my understanding of GitFlow, this should be merged in to `release` and then the bug fixes from release are merged back into development but feel free to change the base branch if that's incorrect.)

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -1881,8 +1881,24 @@ FTLdetect() {
...
 
+make_temporary_log() {
+  # Create a random temporary file for the log
+  local TEMPLOG=$(mktemp /tmp/pihole_temp.XXXXXX)
+  # Open handle 3 for templog
+  # https://stackoverflow.com/questions/18460186/writing-outputs-to-log-file-and-console
+  exec 3>"$TEMPLOG"
+  # Delete templog, but allow for addressing via file handle
+  # This lets us write to the log without having a temporary file on the drive, which
+  # is meant to be a security measure so there is not a lingering file on the drive during the debug process
```
Comment (Mcat12) [inline automated install/basic-install.sh]: The comment says `debug`, should be `install` (simple copy-paste error)

**Comment by DL6ER** [comment]
I have to temporarily change the destination branch from `release` to `development` so I can delete the `release` branch.

**Comment by DL6ER** [comment]
Destination branch is now `release/3.2`

File: automated install/basic-install.sh (hunk 1 above)
Comment (dschaper) [reply automated install/basic-install.sh]: Thanks, will fix.

**Comment by dschaper** [comment]
Thanks for the branch changes. I'm sorry for the delays on this, I've hit some personal issues that pulled me away from Pi-hole development for a bit and I'm trying to catch up and get this pushed out. @ThomasKaiser does this look like it solves the issues you were concerned about?

**Comment by Mcat12** [comment]
Some of these changes are implemented in #1925

**Comment by dschaper** [comment]
This was implemented in that new PR. This one can die off.

**Pull request closed without merging**